nasa-jpl/explorer-1
# Feature: more vue components, dynamic Light and Dark theme variants

### Checklist

  - PR titles are used to generate the change log in [releases](../releases); good ones make that easier to scan.
  - PRs will be broadly categorized in the change log, but for even easier scanning, consider prefixing with a component name or other useful categorization, e.g., "BaseButton: fix layout bug", or "Storybook: Update dependencies".
  - These tags are used to do the aforementioned broad categorization in the change log and determine what the next release's version number should be.
  - [Release Drafter](https://github.com/marketplace/actions/release-drafter) will attempt to do the category labeling for you! Please double-check its work.

## Description

This is a follow-up PR to #433 and addresses some needs in https://github.com/nasa-jpl/www/issues/26:
- Theme colors and light/dark variants are now managed entirely in the TailwindCSS config (default, edu and internal themes)
- Adds ability to switch themes and theme variants in Storybook
- Replaces `theme-color` color in TailwindCSS config with `action` and `emphasis` rules (these labels are somewhat arbitrary and can change) and sets appropriate values for the three themes
- removes Tailwindcss Themer, as we ended up not needing it
- Modifies some theme colors to work with interchangable `light`, `dark`, and `darker` naming conventions. These colors will be updated to match the spec once finalized.
- Adds more base blocks and a couple of streamfield blocks to the Vue library

## Instructions to test

1. `pnpm i`
2. `make vue-storybook`
3. View the BlockQuote stories: http://localhost:6006/?path=/docs/components-blocks-blockquote--docs
4. Change the Theme and Variant options in the tool. Try different combinations.
5. Browse to look at other stories to see similar effects (BlockQuote is the only one that uses EDU's "emphasis" color).

## Tested in the following environments/browsers:



### Operating System


### Browser
